osu.Game.Rulesets.Osu/Beatmaps/OsuBeatmapProcessor.cs

```diff
@@ -41,47 +42,47 @@ public override void PostProcess()
         {
             base.PostProcess();
 
-            var osuBeatmap = (Beatmap<OsuHitObject>)Beatmap;
```

Comment:
This hard cast needs to be gone for the `EditorBeatmap` change to have a chance of working, because `EditorBeatmap` is not a `Beatmap<OsuHitObject>`. The catch processor strangely does not do this and does nice proper soft casts instead.

As an "optimisation" (reducing iterator and list allocations) I added the soft-cast to `List<OsuHitObject>`.